opensrs/opensrsapi.py

```diff
@@ -473,19 +466,37 @@ def get_domains_by_expiredate(self, start_date, end_date, page=None):
         if qpage is None:
             qpage = 1
         while qpage is not None:
-            rsp = self._get_domains_by_expiredate(start_date, end_date, qpage)
-            data = rsp.get_data()
-            for domain in data['attributes']['exp_domains']:
+            data = self._get_domains_by_expiredate(start_date, end_date, qpage)
+            for domain in data['exp_domains']:
                 domains.append({
                     'domain': domain['name'],
                     'domain_expiration': domain['expiredate'],
                 })
-            if data['attributes']['remainder'] == '0':
+            if data['remainder'] == '0':
                 qpage = None
             else:
                 qpage += 1
         return domains
 
+    def iterate_domains(self, expiry_from, expiry_to):
+        pagination_options = {RESULTS_KEY: 'exp_domains'}
+        return PaginatedResults(
+            self._get_domains_by_expiredate, args=(expiry_from, expiry_to),
+            **pagination_options)
+
+    def _get_domains_by_expiredate(self, expiry_from, expiry_to,
+                                   page=1, page_size=40):
```

Comment:
I find this default for `page` parameter quite confusing. Making it obligatory we'll make code less implicit.